packages/server/lambdas/alfresco/list-files.ts

```diff
@@ -104,31 +113,46 @@ export async function handleRequest(event: ALBEvent): Promise<ALBResult> {
     }
     const endpoint = findEndpoint(category, fileEndpointsCache);
     const alfrescoParent = endpoint?.alfrescoFolder;
-    const hasClassifiedContent = endpoint?.hasClassifiedContent;
 
     if (!alfrescoParent) {
       throw new RataExtraLambdaError('Category not found', 404);
     }
 
-    const categoryData = await searchByTermWithParent(user.uid, alfrescoParent, subCategory, categoryPage, language);
-    if (subCategory) {
-      const folderId = get(categoryData, 'list.entries[0].entry.id', -1);
-      const subCategoryData = await searchByTermWithParent(user.uid, folderId, '', page, language);
-      return {
-        statusCode: 200,
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify({ data: subCategoryData, hasClassifiedContent }),
-      };
+    let data;
+    if (!folderId) {
+      data = await searchByTermWithParent(user.uid, alfrescoParent, page, language);
+    } else {
+      const foundFolder = await getFolder(user.uid, folderId);
+      const folderPath = get(foundFolder, 'entry.path.name', '');
+      const isFolderDescendantOfCategory = await isFolderInCategory(folderPath, category);
+      if (isFolderDescendantOfCategory) {
+        data = await searchByTermWithParent(user.uid, folderId, page, language);
+      }
     }
 
+    const responseBody = {
+      hasClassifiedContent: endpoint?.hasClassifiedContent || true,
```

Comment:
I'm not up-to-date with the classified content stuff, but can fallback `true` here lead to many false positives?

Reply:
fallback `true` would just prevent user from manipulating folders/files in the given category context. Removing "true" would probably be better as it was original and would be null otherwise if fallback